### PR TITLE
RDK-28817: Adding opt-in 'PackagerEx' (DAC) extensions ... default: OFF   (CODE ONLY)

### PR DESCRIPTION
**Pull request by FitzerIRL**
NOTE: This PR depends on Thunder PR 323 for interface changes.

This is the implementation ONLY ... does NOT include the Demo app.

Use `cmake  -DBUILD_WITH_PACKAGER_EX=true` to include the new **PackagerEx** code and dependancies in your build.

TODO:  **IPackager.h** interface arrangements will need to be made.

### CODE REVIEW

File: Packager/CMakeLists.txt [hunk 1]
```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
...
+else (BUILD_WITH_PACKAGER_EX)
+
+    message("Building with support for legacy PACKAGER")
+
+    set(PLUGIN_NAME Packager)
+    set(MODULE_NAME ${NAMESPACE}${PLUGIN_NAME})
+
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+    find_path(LIBUUID_INCLUDE_DIR uuid.h PATH_SUFFIXES uuid)
```
Comment (MFransen69) [inline Packager/CMakeLists.txt:137]: If I'm right uuid is not needed for legacy packager support?

File: Packager/CMakeLists.txt [hunk 2]
```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
...
+    find_library(LIBUUID_LIBRARY libuuid.so)
+    include_directories(${LIBUUID_INCLUDE_DIR})
+
+    find_package(${NAMESPACE}Plugins REQUIRED)
+    find_package(libprovision QUIET)
+    find_package(LibOPKG REQUIRED)
+    find_package(GLIB REQUIRED)
```
Comment (MFransen69) [inline Packager/CMakeLists.txt:144]: Think GLIB would not be needed for legacy packager, correct? And same for Curl below

File: Packager/CMakeLists.txt [hunk 3]
```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
...
+
+    find_package(CURL)
+    find_package(CompileSettingsDebug CONFIG REQUIRED)
+
+    # enabling the secure extension requires a license: https://www.hwaci.com/cgi-bin/see-step1
+    if (USE_SQLITE_SEE)
```
Comment (MFransen69) [inline Packager/CMakeLists.txt:150]: SQL lite should not be needed for legacy support I think?

File: Packager/CMakeLists.txt [hunk 4]
```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
...
+    else()
+        find_package(Sqlite REQUIRED)
+    endif()
+
+
+    add_library(${MODULE_NAME} SHARED
+        Module.cpp
+        Packager.cpp
+        PackagerImplementation.cpp
+        PackagerExUtils.cpp
```
Comment (MFransen69) [inline Packager/CMakeLists.txt:161]: Guess you meant to only include the legacy code?

File: Packager/CMakeLists.txt [hunk 5]
```diff
@@ -20,48 +20,207 @@ set(MODULE_NAME WPEFramework${PLUGIN_NAME})
...
+        PackagerExImplementation.cpp)
+
+    if (CURL_FOUND)
```
Comment (MFransen69) [inline Packager/CMakeLists.txt:164]: And if the above is correct I assume all the references to the new dependencies below can also be removed?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Packager/CMakeLists.txt (hunk 3 above)
Comment (FitzerIRL) [reply Packager/CMakeLists.txt:150]: Correct - Thanks !!

File: Packager/CMakeLists.txt (hunk 2 above)
Comment (FitzerIRL) [reply Packager/CMakeLists.txt:144]: Correct - Thanks !!

File: Packager/CMakeLists.txt (hunk 1 above)
Comment (FitzerIRL) [reply Packager/CMakeLists.txt:137]: Correct - Thanks !!

File: Packager/CMakeLists.txt (hunk 5 above)
Comment (FitzerIRL) [reply Packager/CMakeLists.txt:164]: Correct again - Thanks for spotting all these.

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**